test/unit/libp2p_port_test.exs

```diff
@@ -71,4 +71,47 @@ defmodule Unit.Libp2pPortTest do
       bootnodes: bootnodes
     )
   end
+
+  defp retrying_publish(pid, topic, message) do
+    # Give a head start to the other process
+    Process.sleep(1)
+
+    # Publish message
+    :ok = Libp2pPort.publish(pid, topic, message)
+
+    # Receive acknowledgement or retry publish
+    receive do
+      :ok -> :ok
+    after
+      1 -> retrying_publish(pid, topic, message)
+    end
+  end
+
+  test "start two hosts, and gossip about" do
+    gossiper_addr = ["/ip4/127.0.0.1/tcp/48766"]
+    start_port(:publisher)
+    start_port(:gossiper, listen_addr: gossiper_addr)
+
+    topic = "/test/gossipping"
+    message = "hello world!"
+
+    # Connect the two peers
+    id = Libp2pPort.get_id(:gossiper)
+    :ok = Libp2pPort.add_peer(:publisher, id, gossiper_addr, 999_999_999_999)
+
+    pid = self()
+
+    spawn(fn ->
+      # Subscribe to the topic
+      :ok = Libp2pPort.subscribe_to_topic(:gossiper, topic)
+
+      # Receive the message
+      assert {^topic, ^message} = Libp2pPort.receive_gossip()
+
+      # Send acknowledgement
+      send(pid, :ok)
+    end)
```

Comment:
This test is great, thanks for writing it! It's not trivial to test any of this. I'm wondering if we can:
1. Avoid sleeping in the test.
2. Avoid retries (or at least have a maximum amount of retries before it fails).

Maybe we should send a `:ready` message to `pid` after subscribing to the topic instead of sleeping? Once that function returns `ok` we're already subscribed, aren't we?

Reply:
Done 👍